streams/src/main/java/org/apache/kafka/streams/kstream/Predicate.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream;
+
+public interface Predicate<K, V> {
+
+    boolean apply(K key, V value);
```

Comment:
Inserting new class or interface types in the type hierarchy is listed as a [binary-compatible change in Java SE7](https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html) and [Java SE8](https://docs.oracle.com/javase/specs/jls/se8/html/jls-13.html). So if we rename the method now, then if/when `Predicate<K,V>` is changed to extend `java.util.function.BiPredicate<K,V>` and released clients using `Predicate<K,V>` should not have to recompile.